scene/resources/audio_stream_wav.cpp

```diff
@@ -33,6 +33,17 @@
 #include "core/io/file_access_memory.h"
 #include "core/io/marshalls.h"
 
+#define DRWAV_IMPLEMENTATION
+#define DRWAV_NO_STDIO
+#define DR_WAV_LIBSNDFILE_COMPAT
+#define DRWAV_MALLOC(sz) memalloc(sz)
+#define DRWAV_REALLOC(p, sz) memrealloc(p, sz)
+#define DRWAV_FREE(p) \
+	if (p)            \
+	memfree(p)
```

Comment:
@Calinou Do we usually set the thirdparty implementation defines in the engine code or inside the thirdparty itself?

Reply:
I've seen single-header thirdparty implementations being defined on the only class they are being used, like minimp3(https://github.com/godotengine/godot/blob/47bc374edf6d2e775a5e6b937dc3fd73cdc6f59b/modules/minimp3/audio_stream_mp3.cpp#L33).

Reply:
> However, doing this will still prevent using a system-provided library as we expect it to be compiled with certain defines (it's only an issue for native Linux packages).

So basically, this would require a build option like `builtin_dr_libs`? I don't know exactly how builtins are handled.

Reply:
@Calinou

Fedora seems to only provide `dr_wav` as header-only `-devel` package, which only provides a `dr_wav.h` file to be included and implemented on the same way it is now.

I couldn't find other package managers that provide a `dr_wav` package.

I don't think making a `builtin_dr_libs` variable is worth it, considering it wouldn't make a difference after compilation.

Reply:
Indeed, it doesn't really make sense to be able to link against a system library here.